### DISCUSSION TITLE
gatsby-plugin-theme-ui wants theme-ui ^0.5.0-alpha.0

### DISCUSSION THREAD

**Discussion by pszafer** [system-ui/theme-ui#1618] [Q&A] [answered]
Hi,

I'm trying to use Theme-UI, looks great so far, but I think either I'm doing something wrong or peerDependencies are outdated.
Plugin always want to install `"theme-ui": "^0.5.0-alpha.0"`, I forced it to install v0.6.1, but even then I have errors:
```
importSource cannot be set when runtime is classic.
```

Are there any older docs for v0.5.0 as guide to follow? or maybe I'm doing something wrong?

**Comment by hasparus** [comment, marked as answer]
`peerDependencies` were oudated. Thanks for catchng this!

> importSource cannot be set when runtime is classic.

AFAIK Gatsby's Babel config uses classic runtime by default. You'd have to overwrite this setting in your Babel config like this: https://www.gatsbyjs.com/docs/reference/release-notes/v3.0/#babel-preset-gatsby

If you configure Theme UI as your importSource in the config you won't need to use any JSX pragma comment.

Alternatively, you can still use classic JSX runtime and go without modifying Babel config by using `@jsx` pragma comment.

Try this:
```
/** @jsx jsx */
import { jsx } from 'theme-ui';
```

**Reply by pszafer** [reply]
thanks for reply! so I have to use canary version for now, correct?
same bug exists for prism package.

**Reply by hasparus** [reply]
> I have to use canary version for now, correct?

You should be able to disregard the warning without any problems. The bug is not connected to it and should be fixed with a Babel config change.
